### ISSUE TITLE
Mirth fails reading big files, but does not show any problem report

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#1395]
When mirth tries to read a too big file, it fails, but does show no fail notice

Imported Issue. Original Details:
Jira Issue Key: MIRTH-1396
Reporter: albertosaez
Created: 2010-04-05T15:50:20.000-0700

**Comment by rbeckman-nextgen** [comment]
This patch shows the java heap notification error that happens when Mirth can't allocate enough memory to read a file.

Java heap OutOfMemory are throwable objects, but no Exceptions, and this is why the normal catch sentences don't get it.

I've add the same control when the Out Of Memory happens in the JavaScript Preprocessor (where the big DOM objects and strings are created). 

Of course this not solve the problem, but at least the user can realize what has happened.

Imported Comment. Original Details:
Author: albertosaez
Created: 2010-04-05T15:58:34.000-0700

**Comment by rbeckman-nextgen** [comment]
Added throwable catches for out of memory exceptions when reading files.

Imported Comment. Original Details:
Author: jacobb
Created: 2010-04-07T12:00:35.000-0700

**Comment by rbeckman-nextgen** [comment]
Did not see the reason for moving "byte[] contents = getBytesFromFile(file);", so I left that change off.  Let me know if there was another change related to this that you wanted added.

Imported Comment. Original Details:
Author: jacobb
Created: 2010-04-07T12:06:27.000-0700

**Comment by rbeckman-nextgen** [comment]
I changed

byte[] contents = getBytesFromFile(file);
message = new String(getBytesFromFile(file), contents);

To

message = new String(getBytesFromFile(file), fileConnector.getCharsetEncoding());

To prevent JVM duplicating the content of the file in memory if the bytecode optimizing process is not good enough. I made the test with 90Mb files. This means 180Mb in memory and the time in copying the content, so a bad optimization could give a measurable performance loss.


Imported Comment. Original Details:
Author: albertosaez
Created: 2010-04-07T12:43:18.000-0700
